### PR TITLE
[6.3] host collection change assigned content (BZ1315280)

### PR DESCRIPTION
**Pull request by ldjebran**
cover : https://bugzilla.redhat.com/show_bug.cgi?id=1315280
```
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/ui/test_hostcollection.py::HostCollectionPackageManagementTest::test_positive_change_assigned_content
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-29 12:06:51 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_hostcollection.py::HostCollectionPackageManagementTest::test_positive_change_assigned_content PASSED

============================================== 1 passed in 771.26 seconds ==============================================
```

### CODE REVIEW

File: robottelo/ui/hostcollection.py [hunk 1]
```diff
@@ -194,3 +194,36 @@ def execute_bulk_errata_installation(self,
...
         return result.get_attribute('type')
+
+    def change_assigned_content(self, name, lce_name, cv_name, timeout=120):
+        """Change assigned life cycle environment and content view of host
+        collection
+
+        :param name: the host collection name
+        :param cv_name: the content view name to assign
+        :param lce_name: the life cycle environment to assign
+        :param timeout: Timeout in seconds for content change task to finish
```
Comment (oshtaier) [inline robottelo/ui/hostcollection.py]: provide
:return:

File: tests/foreman/ui/test_hostcollection.py [hunk 2]
```diff
@@ -690,8 +690,119 @@ def test_positive_install_errata(self):
...
+               and "All local data refreshed" message is displayed
+            3. The urls listed by last command "subscription-manager repos" was
+               updated to the new Life cycle environment and content view
+               names
+
+        :BZ: 1315280
+
+        :CaseLevel: System
+        """
+        # create a clients variable here as self.hosts attribute will be
```
Comment (oshtaier) [inline tests/foreman/ui/test_hostcollection.py]: And how did other tests work in that class? Maybe it will be better to do rename in setup method?

Comment (ldjebran) [reply tests/foreman/ui/test_hostcollection.py]: other tests user self.hosts before Session

Comment (ldjebran) [reply tests/foreman/ui/test_hostcollection.py]: But this test user self.hosts that are vms in UI session

Comment (ldjebran) [reply tests/foreman/ui/test_hostcollection.py]: I open an issue for this https://github.com/SatelliteQE/robottelo/issues/5329

**Comment by ldjebran** [comment]
@oshtaier comments addressed

File: tests/foreman/ui/test_hostcollection.py [hunk 3]
```diff
@@ -690,8 +690,119 @@ def test_positive_install_errata(self):
...
+        expected_repo_urls = {client.hostname: [] for client in clients}
+        for client in clients:
+            result = client.run("subscription-manager repos")
+            self.assertEqual(result.return_code, 0)
+            client_repo_urls = [
+                line.split(' ')[-1]
+                for line in result.stdout
+                if line.startswith(repo_line_start_with)
+            ]
+            self.assertGreater(len(client_repo_urls), 0)
```
Comment (oshtaier) [inline tests/foreman/ui/test_hostcollection.py]: maybe we can change that assertion on:
```
self.assertEqual(
    set(expected_repo_urls[client.hostname]),
    set(client_repo_urls)
)
```
just to have better visibility that setting was really changed?

Comment (ldjebran) [reply tests/foreman/ui/test_hostcollection.py]: @oshtaier these sets cannot be equal as the expected URLs has been updated by the new lce/cv values, perhaps you mean assertNotEqual?

Comment (ldjebran) [reply tests/foreman/ui/test_hostcollection.py]: @oshtaier any think we need to be sure that there some urls there and add the assert not equal after expected_repo_urls initialization after line 779

Comment (oshtaier) [reply tests/foreman/ui/test_hostcollection.py]: I meant specifically to check before update and after

**Comment by ldjebran** [comment]
@oshtaier PR update please revisit 

- with sat toolls with cdn repo
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/ui/test_hostcollection.py::HostCollectionPackageManagementTest::test_positive_change_assigned_content
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-10-11 12:00:13 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_hostcollection.py::HostCollectionPackageManagementTest::test_positive_change_assigned_content PASSED

============================================== 1 passed in 876.77 seconds ==============================================
```
- with sat toolls with custom repo
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/ui/test_hostcollection.py::HostCollectionPackageManagementTest::test_positive_change_assigned_content
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-10-11 12:14:57 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_hostcollection.py::HostCollectionPackageManagementTest::test_positive_change_assigned_content PASSED

============================================== 1 passed in 812.50 seconds ==============================================
```

**Pull request merged**